pyat/at/lattice/elements/magnet_elements.py

```diff
@@ -168,29 +176,80 @@ def __setattr__(self, key, value):
                 raise ValueError(f"MaxOrder must be smaller than {lmax}")
         super().__setattr__(key, value)
 
+    def _get_order(self):
+        order = getattr(self, 'DefaultOrder', None)
+        if order is None:
+            msg = (f"Default Order not defined for class "
+                   f"{self.__class__.__name__}, please access the strength "
+                   f"with PolynomA/B or use predefined magnet class")
+            raise AtError(msg)
+        return order
+    
+    def _get_strength(self, order, integrated=False):
+        try:
+            k = self.PolynomB[order]
```

Comment:
better:
```
k = float(self.PolynomB[order])
```
which converts from array scalar to float, as it was done before.
The result is the same, but the printout is nicer (avoids `np.float64(0.51)`), and allows to add a type hit to the function: `-> float`.

For the future: it also triggers the evaluation of parameters.